src/frontend/src/components/nav/PartCategoryTree.tsx

```diff
@@ -59,6 +66,56 @@ export function PartCategoryTree({
     );
   }
 
+  const mantineTheme = useMantineTheme();
+  const currentTheme =
+    mantineTheme.colorScheme === 'dark'
+      ? mantineTheme.colorScheme
+      : mantineTheme.primaryColor;
+
+  const themes: ThemeSettings = {
```

Comment:
It would make sense to wrap this into a useMemo for performance reasons.

Reply:
What would the dependency be then? Output of `useMantineTheme()`?

Reply:
Yes, every variable that is used inside the useMemo